components/carbon-transports/websocket/org.wso2.carbon.websocket.transport/src/main/java/org/wso2/carbon/websocket/transport/WebsocketConnectionFactory.java

```diff
@@ -127,9 +127,27 @@ public WebSocketClientHandler cacheNewConnection(final URI uri,
 
             String scheme = uri.getScheme() == null ? WebsocketConstants.WS : uri.getScheme();
             final String host = uri.getHost() == null ? "127.0.0.1" : uri.getHost();
-            final int port = uri.getPort();
-            if (!WebsocketConstants.WS.equalsIgnoreCase(scheme) && !WebsocketConstants.WSS.equalsIgnoreCase(scheme)) {
-                return null;
+            final int port;
+            switch (scheme) {
+                case WebsocketConstants.WS:
+                    if (uri.getPort() < 0) {
+                        // Setting default port for WS connection if port not defined in URI
+                        port = WebsocketConstants.WEBSOCKET_DEFAULT_WS_PORT;
+                        break;
+                    }
+                    port = uri.getPort();
+                    break;
+                case WebsocketConstants.WSS:
+                    if (uri.getPort() < 0) {
+                        // Setting default port for WSS connection if port not defined in URI
+                        port = WebsocketConstants.WEBSOCKET_DEFAULT_WSS_PORT;
+                        break;
+                    }
+                    port = uri.getPort();
+                    break;
+                default:
+                    // If scheme does not belong to either ws or wss schemes, we return null
+                    return null;
```

Comment:
dont we need to throw and error here?

Reply:
It was returning null in the previous implementation. If we are to throw an error here the flow will be changed. So we cannot throw an error.